system-test/cluster.ts

```diff
@@ -0,0 +1,326 @@
+        ]);
+        const cluster: Cluster = instance.cluster(clusterId);
+        await checkMetadata(
+          cluster,
+          {
+            ...createClusterOptions,
+            nodes: minServeNodes,
+          },
+          true
+        );
+      });
+      it('should create an instance and then create clusters for automatic scaling', async () => {
+        const clusterId: string = generateId('cluster');
+        await createStandardNewInstance(clusterId, 2);
+        const clusterId2: string = generateId('cluster');
+        const cluster: Cluster = instance.cluster(clusterId2);
+        await cluster.create(createClusterOptions);
+        await checkMetadata(
+          cluster,
+          {
+            ...createClusterOptions,
+            nodes: minServeNodes,
+          },
+          true
+        );
+      });
+    });
+  });
+  describe('Update cluster', () => {
+    describe('Starting from manual scaling', () => {
+      let cluster: Cluster;
+      const startingNodes = 2;
+
+      beforeEach(async () => {
+        const clusterId = generateId('cluster');
+        await createStandardNewInstance(clusterId, startingNodes);
+        cluster = instance.cluster(clusterId);
+      });
+
+      it('should change nodes for manual scaling', async () => {
+        const updateNodes = 5;
+        assert.notEqual(startingNodes, updateNodes);
+        await cluster.setMetadata({nodes: updateNodes});
+        await checkMetadata(
+          cluster,
+          {
+            nodes: updateNodes,
+          },
+          false
+        );
+      });
+      it('should change cluster to autoscaling', async () => {
+        const minServeNodes = 3;
+        const maxServeNodes = 4;
+        const cpuUtilizationPercent = 50;
+        await cluster.setMetadata({
+          minServeNodes,
+          maxServeNodes,
+          cpuUtilizationPercent,
+        });
+        await checkMetadata(
+          cluster,
+          {
+            nodes: startingNodes,
+            minServeNodes,
+            maxServeNodes,
+            cpuUtilizationPercent,
+          },
+          true
+        );
+      });
+      describe('Using an incorrect configuration', () => {
+        it('should throw an error when providing no cluster configuration', async () => {
+          try {
+            await cluster.setMetadata({});
+            assert.fail();
+          } catch (e) {
+            assert.equal(e.message, ClusterUtils.noConfigError);
+          }
+        });
+        it('should throw an error when providing manual and autoscaling configurations', async () => {
+          try {
+            await cluster.setMetadata({
+              nodes: 2,
+              minServeNodes: 3,
+            });
+            assert.fail();
+          } catch (e) {
+            assert.equal(e.message, ClusterUtils.allConfigError);
+          }
+        });
+        it('should throw an error providing all autoscaling configurations', async () => {
```

Comment:
i think this is the wrong description for this test?

Reply:
Yeah. This is a confusing error. Related to the comment above. Changing this now.